BootloaderCommonPkg/Include/Library/CryptoLib.h

```diff
@@ -48,13 +60,45 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 
 typedef UINT8 HASH_CTX[IPP_HASH_CTX_SIZE];   //IPP Hash context buffer
 
-//
-// RSA verify required key parameter format
-//
+
 typedef struct {
-  UINT32                   Signature;
-  UINT8                    PubKeyData[0];
-} RSA_PUB_KEY;
+  //signature ('P', 'U', 'B', 'K')
+  UINT32                   Identifier;
+
+  //Length of Public Key
+  UINT16                   KeySize;
+
+  //KeyType RSA or ECC
+  KEY_TYPE                 KeyType;
+
+  UINT8                    Rsvd;
+
+  //Pubic key data with KeySize bytes
+  //Contains Modulus(256/384 sized) and PubExp[4]
+  UINT8                    KeyData[0];
+} PUB_KEY_HDR;
+
+typedef struct {
+  //signature Identifier('S', 'I', 'G', 'N')
+  UINT32                   Identifier;
+
+  //Length of signature 2K and 3K in bytes
+  UINT16                   SigSize;
+
+ //PKCSv1.5 or RSA-PSS or ECC
+  SIGN_TYPE                SigType;
+
+  //Hash Alg for signingh SHA256, SHA384
+  HASH_ALG_TYPE            HashAlg;
+
+  //Signature length defined by SigSize bytes
+  UINT8                    Signature[0];
+} SIGNATURE_HDR;
+
+
+#define RSA_SIGNATURE_SIZE         (RSA2048_NUMBYTES + sizeof(SIGNATURE_HDR))
+#define RSA_KEYSIZE_SIZE           (RSA_MOD_SIZE + RSA_E_SIZE + sizeof(PUB_KEY_HDR))

```

Comment:
Given we support RSA2048 and RSA3072, should we use RSA2048_SIGNATURE_SIZE for RSA2048?

Reply:
This would be improved with RSA3072 support. Currently we fixed to RSA2K sizes for this patch

Reply:
OK to update it later.